src/zarr/storage/_fsspec.py

```diff
@@ -281,6 +283,19 @@ async def delete(self, key: str) -> None:
         except self.allowed_exceptions:
             pass
 
+    async def delete_dir(self, prefix: str) -> None:
+        # docstring inherited
+        if not self.supports_deletes:
+            raise NotImplementedError('This method is only available for stores that support deletes.')
+        if not self.supports_listing:
```

Comment:
This is not actually a required condition, we don't call list anymore, it is handled internally

I note that coverage things that the few lines *above* were missed.

Reply:
Looks like the exception text needed to match exactly as in the test, globs don't work